lib/promscrape/scrapework.go

```diff
@@ -616,8 +624,14 @@ func (sw *scrapeWork) processDataInStreamMode(scrapeTimestamp, realTimestamp int
 		}()
 
 		samplesScraped.Add(int64(len(rows)))
-		wc.addRows(cfg, rows, scrapeTimestamp, true)
-
+		if err := wc.tryAddRows(cfg, rows, scrapeTimestamp, true); err != nil {
+			if errors.Is(err, errLabelsLimitExceeded) {
```

Comment:
Currently, the error handling logic only works in streaming mode. I think it should also be applied to `processDataOneShot`.

Reply:
What was the purpose of errLabelsLimitExceeded?

From my perspective, it could be removed entirely, the error message could be constructed directly in the `tryAddRow` function, where all the necessary information is already available. In this case, there is no need to handle the error separately in streaming and one-shot mode.

Reply:
I added it for one reason only: bring up error processing to the same level as `SampleLimit`. To keep the metric increment logic and scrape discard logic close to each other.
Otherwise, it is unclear why `func (wc *writeRequestCtx) tryAddRow` has anything to do with `scrapesSkippedByLabelLimit` metric, as it might be used out of scrape context.